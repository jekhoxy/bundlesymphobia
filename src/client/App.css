input {
  background-color: transparent;
}

.App-logo {
  height: 150px;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

a,
a:visited {
  text-decoration: none;
  color: inherit;
}

.react-icons {
  vertical-align: middle;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

h2,
h3,
h4 {
  color: #989898;
  font-weight: 200;
  text-transform: uppercase;
}
