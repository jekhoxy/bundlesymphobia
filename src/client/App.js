import React from "react";
import { Route, Switch, useLocation } from "react-router-dom";
import "./App.css";
import Main from "./page/main";
import Results from "./page/results";

function useQuery() {
  return new URLSearchParams(useLocation().search);
}

function App() {
  const query = useQuery();

  return (
    <Switch>
      <Route path="/results">
        <Results bundle={query.get("bundle")} version={query.get("version")} />
      </Route>

      <Route path="/">
        <Main />
      </Route>
    </Switch>
  );
}

export default App;
