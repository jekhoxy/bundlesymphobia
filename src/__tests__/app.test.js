import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import App from "../client/app";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const initialState = {
  suggestions: [],
};

const store = mockStore(initialState);

describe("Test app router", () => {
  it("should render main page with search input", () => {
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();

    const { getByPlaceholderText, getByAltText } = component;

    expect(getByPlaceholderText("Find a package")).not.toBeNull();

    expect(getByAltText("logo")).not.toBeNull();
  });

  it("should render results page with bundle info div", () => {
    const component = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/results?bundle=react&version=1.0.0"]}>
          <App />
        </MemoryRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();

    const { getByDisplayValue } = component;

    expect(getByDisplayValue("react@1.0.0")).not.toBeNull();
  });
});
