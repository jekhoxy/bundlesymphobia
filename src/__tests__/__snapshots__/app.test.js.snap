// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app router should render main page with search input 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzpans dRFVDX"
      >
        <div
          class="sc-fzplWN fsfINk"
        >
          <img
            alt="logo"
            class="App-logo"
            src="logo.svg"
          />
          <h3>
            Find the cost of adding a npm package to your bundle
          </h3>
          <div
            class="sc-AxhCb hpcrhp"
          >
            <form
              class="sc-AxhUy lexsms"
            >
              <input
                class="sc-AxgMl kkFbcO"
                placeholder="Find a package"
                type="text"
                value=""
              />
              <div
                class="sc-AxirZ sc-AxiKw kPBqlq"
                font-size="inherit"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </form>
            <div
              class="sc-AxheI deggKy"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzpans dRFVDX"
    >
      <div
        class="sc-fzplWN fsfINk"
      >
        <img
          alt="logo"
          class="App-logo"
          src="logo.svg"
        />
        <h3>
          Find the cost of adding a npm package to your bundle
        </h3>
        <div
          class="sc-AxhCb hpcrhp"
        >
          <form
            class="sc-AxhUy lexsms"
          >
            <input
              class="sc-AxgMl kkFbcO"
              placeholder="Find a package"
              type="text"
              value=""
            />
            <div
              class="sc-AxirZ sc-AxiKw kPBqlq"
              font-size="inherit"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </form>
          <div
            class="sc-AxheI deggKy"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test app router should render results page with bundle info div 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzpans dRFVDX"
      >
        <div
          class="sc-fznJRM dqRTcy"
        >
          <a
            href="/"
          >
            <span
              class="sc-fznWqX kSyWGn"
            >
              Bundle
            </span>
            <span
              class="sc-fzoiQi fPTFeg"
            >
              Symphobia
            </span>
          </a>
        </div>
        <div
          class="sc-fzqARJ cJwDgo"
        >
          <div
            class="sc-AxhCb hpcrhp"
          >
            <form
              class="sc-AxhUy lexsms"
            >
              <input
                class="sc-AxgMl kkFbcO"
                placeholder="Find a package"
                type="text"
                value="react@1.0.0"
              />
              <div
                class="sc-AxirZ sc-AxiKw kPBqlq"
                font-size="inherit"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </form>
            <div
              class="sc-AxheI deggKy"
            />
          </div>
          <div
            class="sc-fzqNqU kpiwel"
          >
            <div
              class="sc-fzpjYC dRqaCv"
            >
              <h2>
                bundle size
              </h2>
              <div
                class="sc-fznxsB kkKlep"
              >
                <div
                  class="sc-fzqBZW lffGOc"
                >
                  <div
                    class="sc-fzqNJr kyoOAI"
                  >
                    <label
                      class="sc-fzoLag fbHOLF"
                    >
                      0
                    </label>
                     
                    <label
                      class="sc-fzoXzr fJYbGO"
                    >
                      kB
                    </label>
                  </div>
                  <h4
                    class="sc-fzoyAV lbcKJG"
                  >
                    initial
                  </h4>
                </div>
                <div
                  class="sc-fzqBZW lffGOc"
                >
                  <div
                    class="sc-fzqNJr kyoOAI"
                  >
                    <label
                      class="sc-fzoLag fbHOLF"
                    >
                      0
                    </label>
                     
                    <label
                      class="sc-fzoXzr fJYbGO"
                    >
                      kB
                    </label>
                  </div>
                  <h4
                    class="sc-fzoyAV lbcKJG"
                  >
                    minified
                  </h4>
                </div>
                <div
                  class="sc-fzqBZW lffGOc"
                >
                  <div
                    class="sc-fzqNJr kyoOAI"
                  >
                    <label
                      class="sc-fzoLag fbHOLF"
                    >
                      0
                    </label>
                     
                    <label
                      class="sc-fzoXzr fJYbGO"
                    >
                      kB
                    </label>
                  </div>
                  <h4
                    class="sc-fzoyAV lbcKJG"
                  >
                    gzipped
                  </h4>
                </div>
              </div>
            </div>
            <div
              class="sc-fzoyTs hpcNUO"
              height="400px"
              width="400px"
            >
              <div
                style="width: 100%; height: 100%;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzpans dRFVDX"
    >
      <div
        class="sc-fznJRM dqRTcy"
      >
        <a
          href="/"
        >
          <span
            class="sc-fznWqX kSyWGn"
          >
            Bundle
          </span>
          <span
            class="sc-fzoiQi fPTFeg"
          >
            Symphobia
          </span>
        </a>
      </div>
      <div
        class="sc-fzqARJ cJwDgo"
      >
        <div
          class="sc-AxhCb hpcrhp"
        >
          <form
            class="sc-AxhUy lexsms"
          >
            <input
              class="sc-AxgMl kkFbcO"
              placeholder="Find a package"
              type="text"
              value="react@1.0.0"
            />
            <div
              class="sc-AxirZ sc-AxiKw kPBqlq"
              font-size="inherit"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </form>
          <div
            class="sc-AxheI deggKy"
          />
        </div>
        <div
          class="sc-fzqNqU kpiwel"
        >
          <div
            class="sc-fzpjYC dRqaCv"
          >
            <h2>
              bundle size
            </h2>
            <div
              class="sc-fznxsB kkKlep"
            >
              <div
                class="sc-fzqBZW lffGOc"
              >
                <div
                  class="sc-fzqNJr kyoOAI"
                >
                  <label
                    class="sc-fzoLag fbHOLF"
                  >
                    0
                  </label>
                   
                  <label
                    class="sc-fzoXzr fJYbGO"
                  >
                    kB
                  </label>
                </div>
                <h4
                  class="sc-fzoyAV lbcKJG"
                >
                  initial
                </h4>
              </div>
              <div
                class="sc-fzqBZW lffGOc"
              >
                <div
                  class="sc-fzqNJr kyoOAI"
                >
                  <label
                    class="sc-fzoLag fbHOLF"
                  >
                    0
                  </label>
                   
                  <label
                    class="sc-fzoXzr fJYbGO"
                  >
                    kB
                  </label>
                </div>
                <h4
                  class="sc-fzoyAV lbcKJG"
                >
                  minified
                </h4>
              </div>
              <div
                class="sc-fzqBZW lffGOc"
              >
                <div
                  class="sc-fzqNJr kyoOAI"
                >
                  <label
                    class="sc-fzoLag fbHOLF"
                  >
                    0
                  </label>
                   
                  <label
                    class="sc-fzoXzr fJYbGO"
                  >
                    kB
                  </label>
                </div>
                <h4
                  class="sc-fzoyAV lbcKJG"
                >
                  gzipped
                </h4>
              </div>
            </div>
          </div>
          <div
            class="sc-fzoyTs hpcNUO"
            height="400px"
            width="400px"
          >
            <div
              style="width: 100%; height: 100%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
